### ISSUE TITLE
Episode numbers over 100 are parsed as Season 1 Episode 00 and so on

### ISSUE THREAD

**Issue by sxales** [jellyfin/jellyfin#1180]
**Describe the bug**
Some series don't have seasons, they ran continuously for longer periods of time and the episodes were simply numbered. Those longer running series (more than 100 episodes) are being organized incorrectly. Episode 100 is seen by Jellyfin as Season 1 Episode 00. Episode 200 is seen as Season 2 Episode 00. So on and so forth. In those cases, episodes before 100 are typically put under the Specials folder which makes finding the first episode very difficult. 

Jellyfin also ignores directories if the files are already organized in seasons but still numbered 1-n where n>100. 

~~It is possible to manually correct this error using the Metadata Manager one episode at a time which is not ideal. ~~ I stand corrected. Even when fixed in the Metadata Manager they show up in both the Specials and Season 1 folders.

**Expected behavior**
I believe they should have been put into a Season Unknown folder as some series do when it can't parse or all put into Season 1.

**Screenshots**
Will add later if necessary

**System (please complete the following information):**
 - OS: Windows 10 Version 1809 Build 17763.379
 - Browser: Firefox 66.0.2
 - Jellyfin Version 10.2.2
 - Reverse proxy: No

**Comment by 97carmine** [comment]
In general, I use [MediaElch](https://github.com/Komet/MediaElch) to edit the metadata and the name of the files and it is displayed correctly and I have not had problems with series with more than 100 episodes.

In my case, the season 17 of One Piece has about 118 and one of those episodes is named like this: _One Piece (1999)\Season 17\S17E118 - The rivals remain firm. The fierce monsters of the New World.mkv_

**Comment by Munkenmo** [comment]
I think Onepiece may be an exception here, as opposed to the rule. Eyeshiled 21 has a similar issue as described in the title.

https://old.reddit.com/r/jellyfin/comments/cmn6h6/i_cant_seem_to_get_anime_working_properly/ew5uony/

It may be something to do with the transition from double digit absolute episodes to triple.

**Comment by oddstr13** [comment]
Related issue with fix in Sonarr, may be worth a look; https://github.com/Sonarr/Sonarr/issues/3117

**Comment by Matt07211** [comment]
This has not been resolved, I just tested with fairly tail which is 175 eps long, The first 99 Eps get recognised as specials, and the remain get recognised as season 1

**Comment by Munkenmo** [comment]
Fairy Tail is working fine for me, what's your file structure?

Mine :

Fairy Tail
- Season 01
  -   Fairy Tail - Ep001 [720p]
  -   Fairy Tail - Ep098 [720p]
- Season 02
  - Fairy Tail - Ep049 [720p]
  - Fairy Tail - Ep096 [720p]
- Season 03
  - Fairy Tail - Ep097 [720p]
  - Fairy Tail - Ep150 [720p]

etc. 

![image](https://user-images.githubusercontent.com/17697879/76139639-f2150680-60b6-11ea-8b0f-f5becb54ca43.png)


![image](https://user-images.githubusercontent.com/17697879/76139619-b5e1a600-60b6-11ea-94b5-457702f74956.png)

**Comment by Matt07211** [comment]
I follow Absolute ordering, based off anidb. So `Fairy Tail` will have its 175 eps, then `Fairy Tail (2014)` will have its 102 eps, etc.

**Comment by t0k4rt** [comment]
Same issue here with black Clover, numbering has been restarted after episode 99
 and one piece, numbering has been restarted after episode 899

**Comment by Matt07211** [comment]
@anthonylavado any chance this issue can be reopened, Saving us having to recreate a whole new issue?

**Comment by dkanada** [comment]
So this seems to be working in certain scenarios now which means the initial implementation is working, but not matching enough filenames. Can anyone still having the issue post either exact filenames (including the folder tree) or logs from a library scan so we can improve the matching?

**Comment by malkavi** [comment]
I'm not sure if this is a convention valid path, but the path is only working for episodes < 100.
The episode 100 is not numbered, and from 101 and above are numbered as duplicates of other episodes. 101 -> 1, 102 -> 2... 130 -> 30

"/media/animes/Black Clover/Season 01/[PuyaSubs!] Black Clover - 100 [720p][19802249].mkv"
...
"/media/animes/Black Clover/Season 01/[PuyaSubs!] Black Clover - 130 [720p][F994F013].mkv"

**Comment by krutburken** [comment]
Hey, I've just stumbled onto this problem and saw this issue. 

Is there a workaround for this? I'm not watching anime personally but I have users that do, and this issue is keeping me from migrating from Plex.

I know that is affecting Black Clover in my library (loops back to ep01 after episode 99), unsure if there are others currently but there might be. I've tried on 10.7.0 and this issue seems to persist there even on a fresh library.

**Comment by adcar** [comment]
@krutburken Editing the metadata in Jellyfin for Black Clover, switching the Display Order to "Absolute" and taking everything out of season folders worked for me.

**Comment by Techno-Disaster** [comment]
Looks like this is still a issue

**Comment by carcinocron** [comment]
I don't know any C#, but it looks like this test doesn't cover episode numbers with 3 digits, e.g. 001, 100, 200, 101, etc..

https://github.com/jellyfin/jellyfin/blob/072526d79872b7a8c174636e4709c62bdb182b95/tests/Jellyfin.Naming.Tests/TV/EpisodePathParserTest.cs

I actually had a lot of trouble identifying the issue at first, I thought there were just duplicates, where the first duplicate was showing an episode that is different from it's description.

I attempted putting them in folders for seasons:

```
# sample filenames
/media/TV/DBZK/S02/Episode 55 There Is Planet Earth, Father! Frieza and King Cold Strike Back!.mkv
/media/TV/DBZK/S03/Episode 104 Gohan the Teacher! Videl's Flying Technique Tutorial.mkv
```

![Screenshot at 2022-09-04 23-35-08](https://user-images.githubusercontent.com/4094542/188355914-ced85415-63d9-4745-adba-91704a57d743.png)
![Screenshot at 2022-09-04 23-35-29](https://user-images.githubusercontent.com/4094542/188355915-e3843a90-0f44-4fe3-9900-58d66c074f1d.png)

I manually edited an incorrect episode number (from 1 to 100) and got a better result, although should have been at the bottom instead of the top after a refresh:

![Screenshot at 2022-09-04 23-41-14](https://user-images.githubusercontent.com/4094542/188356438-faaac3ee-64b5-45e5-b579-7b8e216030bb.png)

**Comment by carcinocron** [comment]
recanning library reverts manually correcting the metadata.

**Comment by carcinocron** [comment]
Using "Display Order: Absolute" and _not_ using extra folders seems to have improved things.

![Screenshot at 2022-09-05 00-36-24](https://user-images.githubusercontent.com/4094542/188361581-6f0fddd0-4c78-430b-a256-80b146b9e982.png)

**Comment by rey130** [comment]
hola ai una solucion  atodos sus problemas estava subiendo ami servidor  la serie de RBD y la promera temporada tiene 215 capitulo   queda asi cundo lo pongo de esta manera Franco y Valeria ¿se casan - [S01][E207]
![Captura de pantalla 2024-01-03 142710](https://github.com/jellyfin/jellyfin/assets/11674265/3104b4ca-ea67-4ea9-88a0-b74bd733f921)

**Comment by fiuncho** [comment]
I'm new in jellyfin and I've faced the same issue, as I read that for some people worked ok and for others it didn't work, I was exploring the code and I found that here https://github.com/jellyfin/jellyfin/blob/072526d79872b7a8c174636e4709c62bdb182b95/tests/Jellyfin.Naming.Tests/TV/EpisodeNumberTests.cs there was examples for 3 digits episodes. But that examples was always preceded by a hyphen. So I tried renaming my episodes like this:

```
before: 001-Black_Clover.mp4
after: Black_Clover-001.mp4
```

And refreshed metadata. With this change finally my episodes was shown the correct info. I tested with dragon ball episodes and it worked too, so I think that it is a possible workaround until the issue is fixed.

**Comment by DeeJack** [comment]
I had the same problem. Usually, I named the files `Episode x`, and there was no issue if 'x' was less than 100. In cases where it was 100 or more, I had to change it to `s01e100`, and that worked as it was supposed to.

**Comment by felix920506** [comment]
should already be fixed in 10.9 when following naming convention

**Comment by JPVenson** [comment]
https://jellyfin.org/docs/general/server/media/shows
